src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+```hocon
+payments-task {
+  action = fetch-user-info
+
+  on._success {
+    action = fetch-payment-providers
+
+    on._success {
+      action = fetch-offers
+
+      on._success {
+        action = te-pebble
+      }
+    }
+  }
+}
+```
+
+We defined four nodes that execute sequentialy one after the other. The first one reuses already defined action called `fetch-user-info`. There is also `fetch-payment-providers` and `fetch-offers` actions that we haven't defined yet. Finally, at the end there is a node with `te-pebble` action. This snippet uses pebble templating engine so we have to define a new action for rendering it. If it used handlebars we could reuse our `te-hbs` action.
+
+New actions we have to define are very similar to what we already have. Both fetching actions have `http` factory that calls mocks and `te-pebble` is referencing a knot defined on the event bus. In practice [HOCON](https://github.com/lightbend/config/blob/master/HOCON.md) (the language we are writing the configuration in) allows us to use variables and other means of avoiding copy and paste. For this tutorial we'll keep it simple though. This is the needed configuration:
+
+```hocon
+fetch-payment-providers {
+  factory = http
+  config.endpointOptions {
+    path = /payments
+    domain = webapi
+    port = 8080
+  }
+}
+fetch-offers {
+  factory = http
+  config.endpointOptions {
+    path = /offers
+    domain = webapi
+    port = 8080
+  }
+}
+te-pebble {
+  factory = knot
+  config.address = knotx.knot.te.pebble
+}
+```
+
+Nothing we haven't seen already. When we load the page now we should see that the second snippet is filled with data. In the Knot.x extension we still see two fragments but this time they both have green boxes next to them - both were processed and ended with success. Click the second fragment on the list to view its details:
+
+![Naive Approach](/img/blog/chrome-extension/naive-second-snippet.png)
+
+It works, but it's not the best approach. Right now the `fetch-offers` node waits for the `fetch-payment-providers` to succeed. There is no reason for that (remmeber, we assumed that those pieces of information can be retrieve intependently). There would also be a problem if `fetch-payment-providers` failed for some reason. Then `fetch-offers` would not execute, even if it would be able to retrieve data and show them to the user. It'd be more efficient and safer if those two nodes could execute in parallel.
+
+There are two types of nodes. Everything you've seen up to this point were so called "single nodes". They just execute an action. The second type is a composite node. This kind of node contains an array of other nodes. It executes them in parallel and waits for all of them to complete. If all of them succeed then the composite succeeds. Otherwise the composite fails. This is how we can reorder our nodes to make use of a composite:

```

Comment:
> This kind of node contains an array of other nodes.

This kind of node contains an array of other root nodes (in fact, those are subgraphs).